service/history/historyEngine.go

```diff
@@ -2567,15 +2569,26 @@ func (e *historyEngineImpl) ScheduleDecisionTask(ctx context.Context, scheduleRe
 		RunId:      scheduleRequest.WorkflowExecution.RunId,
 	}
 
-	return e.updateWorkflowExecution(ctx, domainID, execution, false, true,
-		func(msBuilder mutableState, tBuilder *timerBuilder) ([]persistence.Task, error) {
+	return e.updateWorkflowExecutionWithAction(ctx, domainID, execution,
+		func(msBuilder mutableState, tBuilder *timerBuilder) (*updateWorkflowAction, error) {
 			if !msBuilder.IsWorkflowExecutionRunning() {
 				return nil, ErrWorkflowCompleted
 			}
 
-			// Noop
+			postActions := &updateWorkflowAction{
+				createDecision: true,
+			}
+
+			cronBackoffDuration := msBuilder.GetCronBackoffDuration()
+			if msBuilder.GetNextEventID() == int64(2) && cronBackoffDuration != cron.NoBackoff {
```

Comment:
Having a check based on event id makes me little worried.  Can you instead add another field on request for AddDecisionTask API to indicate if this is the first decision from caller perspective.  I think it is much safer to let caller decide if this is the first decision and should we backoff and create workflow backoff timer.

Reply:
Makes sense. Will do.